File: EventTime/src/main/java/com/hakankuru/EventTime/dto/AdminUserDto.java
package com.hakankuru.EventTime.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AdminUserDto {
    private Long userId;
    private String name;
    private String email;
    private String role;         // USER / ADMIN / SUPER_ADMIN
    private Long universityId;   // null ise USER veya SUPER_ADMIN
}

File: eventhub/app/src/main/java/com/hakankuru/eventhub/data/remote/response/AdminUserDto.kt
package com.hakankuru.eventhub.data.remote.response

/**
 * Backend /api/v1/admin/users endpoint'inden dönen kullanıcı modeli.
 * universityId: ADMIN ise bağlı üniversite, USER/SUPER_ADMIN için null.
 */
data class AdminUserDto(
    val userId: Long,
    val name: String,
    val email: String,
    val role: String,          // "USER" | "ADMIN" | "SUPER_ADMIN"
    val universityId: Long?    // null → USER veya SUPER_ADMIN
)
